src/sql/parts/query/execution/queryActions.ts

```diff
@@ -417,6 +418,55 @@ export class ListDatabasesAction extends QueryTaskbarAction {
 	}
 }
 
+/**
+ * Action class that parses the query string in the current SQL text document.
+ */
+export class ParseSyntaxAction extends QueryTaskbarAction {
+
+	public static EnabledClass = '';
+	public static ID = 'parseQueryAction';
+
+	constructor(
+		editor: QueryEditor,
+		@IConnectionManagementService connectionManagementService: IConnectionManagementService,
+		@IQueryManagementService private _queryManagementService: IQueryManagementService,
+		@INotificationService private _notificationService: INotificationService,
+	) {
+		super(connectionManagementService, editor, ParseSyntaxAction.ID, ParseSyntaxAction.EnabledClass);
+		this.enabled = true;
+		this.label = nls.localize('parseSyntaxLabel', 'Parse');
```

Comment:
how will this display in Command Palette?  It should say something like `Query: Parse` or `Parse Query` or something like that.  If it just says `Parse` I think this is too general.